### DISCUSSION TITLE
Solver: out of memory

### DISCUSSION THREAD

**Discussion by MahavirSingh20** [blue-marble/gridpath#1293] [Q&A] [answered]
While running a model with approximately 196 million non zero values in GridPath 0.9 (Python Version 3.9.13) on a system with 128 GB ram following error was received:
Using license file C:\Users\admin1\gurobi.lic
Read LP format model from file C:\Users\admin1\AppData\Local\Temp\tmpt_wpctwk.pyomo.lp
Reading time = 429.63 seconds
x47852276: 72619651 rows, 47852276 columns, 196612921 nonzeros
Changed value of parameter QCPDual to 1
   Prev: 0  Min: 0  Max: 1  Default: 0
Changed value of parameter Threads to 4
   Prev: 0  Min: 0  Max: 1024  Default: 0
Gurobi Optimizer version 9.0.3 build v9.0.3rc0 (win64)
Optimize a model with 72619651 rows, 47852276 columns and 196612921 nonzeros
Model fingerprint: 0x0456a20c
Coefficient statistics:
  Matrix range     [3e-08, 1e+02]
  Objective range  [1e+03, 8e+13]
  Bounds range     [0e+00, 0e+00]
  RHS range        [5e-07, 6e+06]
Warning: Model contains large objective coefficients
         Consider reformulating model or setting NumericFocus parameter
         to avoid numerical issues.

Concurrent LP optimizer: dual simplex and barrier
Showing barrier log only...

Traceback (most recent call last):
  File "<stdin>", line 5, in <module>
  File "C:\Users\admin1\AppData\Local\Programs\Python\Python39\lib\site-packages\pyomo\solvers\plugins\solvers\GUROBI_RUN.py", line 114, in gurobi_run
    model.optimize()
  File "model.pxi", line 843, in gurobipy.Model.optimize
gurobipy.GurobiError: Out of memory
ERROR: Solver (gurobi) returned non-zero return code (1)
ERROR: See the solver log above for diagnostic information.
ERROR:root:Solver (gurobi) did not exit normally

********************************************

**Comment by johnny-eee** [comment]
hi, i am also facing the memory insufficient issue, do u figure it out ? many thanks!

**Comment by PeterKlein11** [comment, marked as answer]
The log says that solver is running in concurrent mode, which means both simplex and barrier are trying to solve at the same time on a large model. Switching to barrier only should work.
